### ISSUE TITLE
Handling uncommitted modification

### ISSUE THREAD

**Issue by mrchief** [peter-evans/create-pull-request#466]
### Subject of the issue

Seems like modified files are not auto committed
![image](https://user-images.githubusercontent.com/781818/88488862-5c432080-cf5e-11ea-8f85-aa341a2b83a0.png)

### Steps to reproduce

```
      - name: Update Changelog.md
        run: |
          echo "Some changes" | cat - ./CHANGELOG.md > ./temp
          mv ./temp ./CHANGELOG.md
          git status

      - name: Create Pull Request
        id: cpr
        uses: peter-evans/create-pull-request@v3
        with:
          token: ${{ secrets.REPO_ACCESS_TOKEN }}
          commit-message: "chore: Update changelog 📚"
          committer: GitHub <noreply@github.com>
          author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
          branch: ${{ github.ref }}
          title: "chore: Automated PR"
          body: |
            Automated Pull Request 
          labels: |
            release
            automated pr
```

After running everything succeeds but no PR is created.

**Comment by peter-evans** [comment]
Hi @mrchief 

The reason this happens is because you've set `branch: ${{ github.ref }}`. What that means is that the `base` of the pull request (the checked out ref) and the branch that you want to create (again, the checked out ref) are the same. Because they are the same, there is no diff and no pull request is created. For more detail about how this action works, see the documentation [here](https://github.com/peter-evans/create-pull-request/blob/master/docs/concepts-guidelines.md#how-the-action-works) and the readme section on [action behaviour](https://github.com/peter-evans/create-pull-request#action-behaviour).

In short, the `branch` input should not be set to an existing branch. It should be the name of a branch that will be created by the action. To fix your workflow change the `branch` input to a fixed name, e.g. `branch: my-pr-branch`.

**Comment by mrchief** [comment]
Thanks for the detailed explanation @peter-evans. I found the same browsing through docs & issues just a while ago. In my case, I'm not using a repo dispatch. My flow is `user creates a release branch > action updates changelog.md > action creates a PR`.

Because I'm operating on an existing branch, I used github script action to then just create a PR:

```
      - name: Create Pull Request
        uses: actions/github-script@v2
        with:
          github-token: ${{secrets.GITHUB_TOKEN}}
          script: |
            const commits = await github.pulls.create({
              owner: context.repo.owner,
              repo: context.repo.repo,
              title: 'chore: Release ${{ env.NEXT_RELEASE_NAME }}',
              head: '${{ github.ref }}'.replace('refs/heads/', ''),
              base: 'develop',
              body: '...',
              maintainer_can_modify: true
            })
```

Thanks for this amazing action though. Hopefully I'll get to use it in future! ✌
